src/components/authorization/authorization.service.ts

```diff
@@ -104,6 +109,37 @@ export class AuthorizationService {
     return parseSecuredProperties(props, permissions, resource.SecuredProps);
   }
 
+  async verifyCanEditChanges<
+    TObject extends Resource,
+    Key extends keyof TObject & string
+  >(
+    baseNode: TObject,
+    props: readonly Key[],
+    changes: { [Key in keyof TObject]?: UnwrapSecured<TObject[Key]> }
+  ) {
+    for (const prop of props) {
+      if (
+        changes[prop] === undefined ||
+        unwrapSecured(baseNode[prop]) === changes[prop]
+      ) {
+        continue;
+      }
+      await this.verifyCanEdit(baseNode, prop);
+    }
+  }
+
+  async verifyCanEdit<
+    DbNode extends Record<string, any>,
+    Key extends keyof DbNode
+  >(baseNode: DbNode, prop: Key) {
+    if (isSecured(baseNode[prop]) && !baseNode[prop].canEdit) {
+      throw new UnauthorizedException(
+        'You do not have permission to update this property',
+        `${baseNode.__typename as string}.${prop as string}`
```

Comment:
This isn't going to work. This is a path from the gql input object to the property. i.e. `user.timezone`. `user` comes from [here](https://github.com/SeedCompany/cord-api-v3/blob/5f0c861ebd5ee9ba00e0db0540e38b2a1141616e/src/components/user/dto/update-user.dto.ts#L58).

We could camelcase the `resource.name` which would work in most cases by convention. But it will still break in a few like LanguageEngagement's firstScripture which would be `engagement.firstScripture` instead of `languageEngagement.firstScripture`.